process/nodeGroupProcessor.go

```diff
@@ -224,24 +228,48 @@ func (hbp *NodeGroupProcessor) StartCacheUpdate() {
 	}(ctx)
 }
 
-func (hbp *NodeGroupProcessor) handleHeartbeatCacheUpdate() {
-	hbts, err := hbp.getHeartbeatsFromApi()
+func (ngp *NodeGroupProcessor) handleHeartbeatCacheUpdate() {
+	hbts, err := ngp.getHeartbeatsFromApi()
 	if err != nil {
 		log.Warn("heartbeat: get from API", "error", err.Error())
 	}
 
 	if hbts != nil {
-		err = hbp.cacher.StoreHeartbeats(hbts)
+		err = ngp.cacher.StoreHeartbeats(hbts)
 		if err != nil {
 			log.Warn("heartbeat: store in cache", "error", err.Error())
 		}
 	}
 }
 
+// GetWaitingEpochsLeftForPublicKey returns the number of epochs left for the public key until it becomes eligible
+func (ngp *NodeGroupProcessor) GetWaitingEpochsLeftForPublicKey(publicKey string) (*data.WaitingEpochsLeftApiResponse, error) {
+	observers, err := ngp.proc.GetAllObservers()
+	if err != nil {
```

Comment:
check for nil/empty publicKey before trying to get all observers?

Reply:
pushed